src/btree/bt_page.c

```diff
@@ -632,6 +632,18 @@ __inmem_row_leaf(WT_SESSION_IMPL *session, WT_PAGE *page)
             tombstone->prepare_state = WT_PREPARE_INPROGRESS;
             F_SET(tombstone, WT_UPDATE_PREPARE_RESTORED_FROM_DS);
             F_SET(upd, WT_UPDATE_RESTORED_FROM_DS);
+
+            /*
+             * If the update and tombstone both are from the same transaction, set the update also
+             * as prepare in-progress.
+             */
+            if (upd->durable_ts == tombstone->durable_ts && upd->start_ts == tombstone->start_ts &&
+              upd->txnid == tombstone->txnid) {
+                upd->durable_ts = WT_TS_NONE;
```

Comment:
Can we change this to an assert?

Reply:
I didn't understand your point clearly. What do you want to assert here?

Reply:
WT_ASSERT(session, upd->durable_ts == WT_TS_NONE)? We should have written 0 to the disk, is that correct?

Reply:
No. we write the durable timestamp also with prepare timestamp to disk and mark it as prepare.

Reply:
OK, got it.